roles/dnsmasq/tasks/manage_host.yml

```diff
@@ -14,72 +14,100 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+- name: Ensure we have the right data and type
+  ansible.builtin.assert:
+    that:
+      - cifmw_dnsmasq_dhcp_entries is defined
+      - (cifmw_dnsmasq_dhcp_entries | type_debug) == 'list'
+    quiet: true
+    msg: >-
+      Error: cifmw_dnsmasq_dhcp_entries must be a dict. Check
+      documentation for proper format.
+
 - name: Assert we have needed host data
   ansible.builtin.assert:
     quiet: true
     that:
-      - cifmw_dnsmasq_host_network is defined
-      - cifmw_dnsmasq_host_state is defined
-      - cifmw_dnsmasq_host_state in ['present', 'absent']
-      - cifmw_dnsmasq_host_mac is defined
-      - cifmw_dnsmasq_host_ips is defined
+      - (entry | type_debug) == 'dict'
```

Comment:
```suggestion
      - entry is mapping
```

as above

Reply:
so here, the replacement isn't that straightforward:

`entry is not string and entry is iterable and entry is mapping` according to their doc
which, here again, is pretty not nice -.-